packages/broker/src/StorageNodeRegistry.ts

```diff
@@ -50,7 +57,7 @@ export class StorageNodeRegistry {
                 return item.storageNodeAddress
             })
             if (addresses.length > 0) {
-                return addresses[Math.floor(Math.random() * addresses.length)]
+                return addresses
             } else if (addresses.length === 0) {
                 return undefined
```

Comment:
Consider replacing with `return addresses` thereby simplifying method signature return type to be `Promise<string[]>` (and having empty array be a natural signal of "no addresses assigned"). User of method then doesn't need to deal with possible undefined check either, which is nice.